src/swap.rs

```diff
@@ -21,33 +24,43 @@ pub async fn hbit_herc20<A, B>(
     herc20_params: herc20::Params,
 ) -> anyhow::Result<()>
 where
-    A: hbit::Fund + herc20::RedeemAsAlice + hbit::Refund,
-    B: herc20::Deploy + herc20::Fund + hbit::RedeemAsBob + herc20::Refund,
+    A: Do<hbit::CorrectlyFunded>
```

Comment:
First read: `Do` and `Execute` verbs are synonyms so it's confusing.

Reply:
Yeah, I haven't found good names for this yet ;(